jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2310,7 +2310,7 @@ public Part getPart(String name) throws IOException, ServletException
         return getParts(null);
     }
 
-    private Collection<Part> getParts(MultiMap<String> params) throws IOException
+    private synchronized Collection<Part> getParts(MultiMap<String> params) throws IOException
```

Comment:
As per offline discussion, lets not insert synchronization into the request, which is not synchronized for any other method - the spec does not have any support for thread safety in calling methods on requests.